### PR TITLE
fix(objects): add public_jobs attribute to projects

### PR DESCRIPTION
**Pull request by nejch**
This will replace the deprecated public_builds attribute. Closes https://github.com/python-gitlab/python-gitlab/issues/1428

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
Looks good to me. Is there a standard and hopefully easy way to add tests for when we add attributes?

**Review by JohnVillalovos — APPROVED** [review]

File: gitlab/v4/objects/users.py [hunk 1]
```diff
@@ -443,6 +443,7 @@ class UserProjectManager(ListMixin, CreateMixin, RESTManager):
             "description",
             "builds_enabled",
             "public_builds",
+            "public_jobs",
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/users.py:446]: Is this the API for this?  https://docs.gitlab.com/ee/api/projects.html#create-project-for-user

I'm guessing a bug in the API docs as they only list `public_builds` for that call.

Comment (JohnVillalovos) [reply gitlab/v4/objects/users.py:446]: Answering my own question:

https://gitlab.com/gitlab-org/gitlab/-/blob/5f16f0762a05506be7fd11760f0c8d82ad0068d8/lib/api/entities/project.rb#L92

Looks like all `public_builds` now have an alias of `public_jobs`

**Comment by JohnVillalovos** [comment]
This commit confuses me: https://gitlab.com/gitlab-org/gitlab/-/commit/38c3791e0ae42fbee9f6a04f31f8b1e9c3056c21

> When creating or updating a user, use public_builds attribute rather than public_jobs attribute in API.

**Comment by nejch** [comment]
> This commit confuses me: https://gitlab.com/gitlab-org/gitlab/-/commit/38c3791e0ae42fbee9f6a04f31f8b1e9c3056c21
> 
> > When creating or updating a user, use public_builds attribute rather than public_jobs attribute in API.
> 
> Found the MR associated with that commit:
> https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/23194

Huh, that's right, thanks. GET responds with `public_jobs`, but `POST/PUT` only take `public_builds` :facepalm: 

Otherwise the whole "build" naming has been deprecated since 9.0, you can't even reach their docs for this anymore - here's 12.10: https://docs.gitlab.com/12.10/ee/ci/variables/deprecated_variables.html. On the other hand, `public_builds` is used all over the codebase, so I don't know how this confusion still drags into 14.0...

I've opened an issue upstream at https://gitlab.com/gitlab-org/gitlab/-/issues/329725 and I'll close this PR for now, since it won't help us much and seems like it would be a bit of work on the gitlab side to change all of this:

```
./app/views/projects/settings/ci_cd/_form.html.haml:            = f.check_box :public_builds, { class: 'form-check-input' }
./app/views/projects/settings/ci_cd/_form.html.haml:            = f.label :public_builds, class: 'form-check-label' do
./app/graphql/types/project_type.rb:    field :public_jobs, GraphQL::BOOLEAN_TYPE, method: :public_builds, null: true,
./app/controllers/projects_controller.rb:      :public_builds,
./app/controllers/projects/settings/ci_cd_controller.rb:          :build_timeout_human_readable, :build_coverage_regex, :public_builds,
./app/policies/project_policy.rb:  condition(:public_builds, scope: :subject, score: 0) { project.public_builds? }
./app/policies/project_policy.rb:  rule { public_builds }.policy do
./app/policies/project_policy.rb:  rule { public_builds & can?(:guest_access) }.policy do
./spec/finders/ci/jobs_finder_spec.rb:  let_it_be(:project) { create(:project, :private, public_builds: false) }
./spec/lib/gitlab/cycle_analytics/permissions_spec.rb:  let(:project) { create(:project, public_builds: false) }
./spec/lib/gitlab/ci/status/build/common_spec.rb:        project.update!(public_builds: false)
./spec/lib/gitlab/import_export/safe_model_attributes.yml:- public_builds
./spec/presenters/ci/build_presenter_spec.rb:      let(:project) { create(:project, public_builds: false) }
./spec/requests/api/merge_requests_spec.rb:        project = create(:project, public_builds: false)
./spec/requests/api/jobs_spec.rb:    create(:project, :repository, public_builds: false)
./spec/requests/api/jobs_spec.rb:            project.update_column(:public_builds, true)
./spec/requests/api/jobs_spec.rb:            project.update_column(:public_builds, true)
./spec/requests/api/jobs_spec.rb:              project.update_column(:public_builds, true)
./spec/requests/api/jobs_spec.rb:            project.update_column(:public_builds, false)
./spec/requests/api/jobs_spec.rb:            project.update_column(:public_builds, true)
./spec/requests/api/jobs_spec.rb:            project.update_column(:public_builds, true)
./spec/requests/api/jobs_spec.rb:      let(:public_builds) { true }
./spec/requests/api/jobs_spec.rb:                       public_builds: public_builds)
./spec/requests/api/jobs_spec.rb:          let(:public_builds) { true }
./spec/requests/api/jobs_spec.rb:          let(:public_builds) { false }
./spec/requests/api/jobs_spec.rb:          let(:public_builds) { true }
./spec/requests/api/jobs_spec.rb:          let(:public_builds) { true }
./spec/requests/api/jobs_spec.rb:        let(:public_builds) { true }
./spec/requests/api/jobs_spec.rb:      where(:public_builds, :user_project_role, :expected_status) do
./spec/requests/api/jobs_spec.rb:          project.update!(public_builds: public_builds)
./spec/requests/api/ci/pipelines_spec.rb:        project.update!(public_builds: false)
./spec/requests/api/ci/pipelines_spec.rb:        project.update!(public_builds: false)
./spec/requests/api/ci/pipeline_schedules_spec.rb:  let_it_be(:project) { create(:project, :repository, public_builds: false) }
./spec/requests/api/ci/pipeline_schedules_spec.rb:      let_it_be(:project) { create(:project, :repository, :public, public_builds: false) }
./spec/requests/api/project_attributes.yml:    public_builds: public_jobs
./spec/requests/projects/cycle_analytics_events_spec.rb:  let(:project) { create(:project, :repository, public_builds: false) }
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: false)
./spec/features/security/project/private_access_spec.rb:  let_it_be(:project, reload: true) { create(:project, :private, :repository, public_builds: false) }
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/private_access_spec.rb:        project.public_builds = false
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: true)
./spec/features/security/project/private_access_spec.rb:        project.update!(public_builds: false)
./spec/features/merge_request/user_sees_merge_widget_spec.rb:        public_builds: false
./spec/features/dashboard/projects_spec.rb:        project.update!(public_builds: false)
./spec/features/projects/jobs/permissions_spec.rb:        project.update!(public_builds: false)
./spec/features/projects/jobs/permissions_spec.rb:        project.update!(public_builds: true)
./spec/features/projects/jobs/permissions_spec.rb:          project.update!(public_builds: false)
./spec/features/projects/jobs/permissions_spec.rb:      where(:public_builds, :user_project_role, :ci_debug_trace, :expected_status_code) do
./spec/features/projects/jobs/permissions_spec.rb:          project.update!(public_builds: public_builds)
./spec/features/projects/jobs/permissions_spec.rb:      where(:public_builds, :user_project_role, :ci_debug_trace, :expected_status_code, :expected_msg) do
./spec/features/projects/jobs/permissions_spec.rb:          project.update!(public_builds: public_builds)
./spec/features/projects/user_sees_sidebar_spec.rb:  let(:project) { create(:project, :private, public_builds: false, namespace: user.namespace) }
./spec/features/projects/user_sees_sidebar_spec.rb:      project.public_builds = true
./spec/features/projects/pipelines/pipeline_spec.rb:        let(:project) { create(:project, :public, :repository, public_builds: false) }
./spec/features/projects/pipelines/pipeline_spec.rb:      project.update!(public_builds: false)
./spec/features/projects/pipelines/pipeline_spec.rb:        project.update!(public_builds: false)
./spec/features/commits_spec.rb:            public_builds: false)
./spec/controllers/projects/graphs_controller_spec.rb:          project.update_column(:public_builds, false)
./spec/controllers/projects/badges_controller_spec.rb:        project.update!(public_builds: true)
./spec/controllers/projects/badges_controller_spec.rb:        project.update!(public_builds: false)
./spec/support/shared_examples/requests/api/pipelines/visibility_table_shared_examples.rb:      public_builds: public_builds
./spec/support/shared_examples/requests/api/pipelines/visibility_table_shared_examples.rb:  where(:visibility_level, :builds_access_level, :public_builds, :is_admin, :user_role, :response_status) do
./spec/support/shared_examples/policies/project_policy_shared_examples.rb:    project.update!(public_builds: false)
./spec/support/shared_examples/policies/project_policy_shared_examples.rb:        project.update!(public_builds: false)
./spec/support/shared_examples/policies/project_policy_shared_examples.rb:        project.update!(public_builds: false)
./spec/support/shared_examples/policies/project_policy_shared_examples.rb:        project.update!(public_builds: false)
./spec/fixtures/lib/gitlab/import_export/designs/project.json:   "public_builds":true,
./spec/fixtures/lib/gitlab/import_export/multi_pipeline_ref_one_external_pr/project.json:    "public_builds": true,
./spec/fixtures/lib/gitlab/import_export/multi_pipeline_ref_one_external_pr/tree/project.json:{"id":5,"approvals_before_merge":0,"archived":false,"auto_cancel_pending_pipelines":"enabled","autoclose_referenced_issues":true,"build_allow_git_fetch":true,"build_coverage_regex":null,"build_timeout":3600,"ci_config_path":null,"delete_error":null,"description":"Vim, Tmux and others","disable_overriding_approvers_per_merge_request":null,"external_authorization_classification_label":"","external_webhook_token":"D3mVYFzZkgZ5kMfcW_wx","public_builds":true,"shared_runners_enabled":true,"visibility_level":20}
./spec/services/notification_service_spec.rb:            project.update!(public_builds: false)
./spec/services/projects/update_service_spec.rb:        result = update_project(project, admin, public_builds: true)
./spec/services/projects/update_service_spec.rb:        expect(project.reload.public_builds).to be true
./spec/policies/ci/build_policy_spec.rb:      project.update_attribute(:public_builds, false)
./lib/api/entities/project.rb:      expose :public_builds, as: :public_jobs
./lib/api/helpers/projects_helpers.rb:        optional :public_builds, type: Boolean, desc: 'Perform public builds'
./lib/api/helpers/projects_helpers.rb:          :public_builds,
./db/structure.sql:    public_builds boolean DEFAULT true NOT NULL,
./db/migrate/20181228175414_init_schema.rb:      t.boolean "public_builds", default: true, null: false
./config/pseudonymizer.yml:    - public_builds
./doc/api/projects.md:| `public_builds`                                             | boolean | **{dotted-circle}** No | If `true`, jobs can be viewed by non-project members. |
./doc/api/projects.md:| `public_builds`                                             | boolean | **{dotted-circle}** No | If `true`, jobs can be viewed by non-project-members. |
./doc/api/projects.md:| `public_builds`                                             | boolean        | **{dotted-circle}** No | If `true`, jobs can be viewed by non-project members. |
./ee/spec/requests/api/jobs_spec.rb:    create(:project, :repository, public_builds: false)
./ee/spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: true)
./ee/spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: false)
./ee/spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: true)
./ee/spec/features/security/project/public_access_spec.rb:        project.update!(public_builds: false)
./ee/spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: true)
./ee/spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: false)
./ee/spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: true)
./ee/spec/features/security/project/internal_access_spec.rb:        project.update!(public_builds: false)

```

**Comment by JohnVillalovos** [comment]
Yeah, it is a bit confusing to me. Though it appears the `public_jobs` is an alias for `public_builds`. My guess is that it would work. But since their API documentation only talks about `public_builds` for the `POST/PUT` methods it makes sense for us to follow the API documentation.

**Pull request closed without merging**